src/main/java/org/openstreetmap/atlas/checks/commands/FlagStatisticsSubCommand.java

```diff
@@ -0,0 +1,619 @@
+    private static final String OUTPUT_TYPES_OPTION = "output-types";
+    private static final String GENERATOR = "generator";
+    private static final String CHECK = "Check";
+
+    private static final Logger logger = LoggerFactory.getLogger(ConfigurationResolver.class);
+
+    private final Gson gson = new Gson();
+    private final OptionAndArgumentDelegate optionAndArgumentDelegate;
+    private final CommandOutputDelegate outputDelegate;
+
+    public static void main(final String[] args)
+    {
+        new FlagStatisticsSubCommand().runSubcommandAndExit(args);
+    }
+
+    public FlagStatisticsSubCommand()
+    {
+        this.optionAndArgumentDelegate = this.getOptionAndArgumentDelegate();
+        this.outputDelegate = this.getCommandOutputDelegate();
+    }
+
+    @Override
+    @SuppressWarnings("squid:S3655")
+    public int execute()
+    {
+        // Read the main input folder
+        final Map<String, Map<String, Counter>> inputCounts = this.getCountryCheckCounts(
+                this.optionAndArgumentDelegate.getOptionArgument(INPUT_OPTION).get());
+
+        // Get the output folder path
+        final String outputFolder = this.optionAndArgumentDelegate.getOptionArgument(OUTPUT_OPTION)
+                .get();
+        // Get the output types
+        final List<String> outputTypes = Arrays
+                .asList(StringUtils.split(this.optionAndArgumentDelegate
+                        .getOptionArgument(OUTPUT_TYPES_OPTION).orElse("full"), ','));
+
+        try
+        {
+            // Write the counts for the input logs is requested
+            if (outputTypes.contains("full"))
+            {
+                this.writeCSV(outputFolder + "/input.csv", generateFullOutput(inputCounts));
+            }
+            // Generate the totals output
+            List<List<String>> totalsOutput = generateTotalsOutputStage1(inputCounts);
+            // Generate the counts output if requested
+            List<List<String>> countsOutput = generateCountsOutputStage1(inputCounts);
+
+            // Get the optional reference input
+            final Optional<String> referenceFilePath = this.optionAndArgumentDelegate
+                    .getOptionArgument(REFERENCE_OPTION);
+            // If a second input is supplied...
+            if (referenceFilePath.isPresent())
+            {
+                // Read the second input
+                final Map<String, Map<String, Counter>> referenceCounts = this
+                        .getCountryCheckCounts(referenceFilePath.get());
+
+                // Get the difference
+                final Map<String, Map<String, Counter>> differenceCounts = this
+                        .getDifference(referenceCounts, inputCounts);
+
+                // Write outputs for the reference and difference if requested
+                if (outputTypes.contains("full"))
+                {
+                    this.writeCSV(outputFolder + "/reference.csv",
```

Comment:
I don't think this output is necessary. This will be a duplicate of the "input.csv" output if we continuously output each `output-types` from run to run